lib/trip_dispatcher.rb

```diff
@@ -1,40 +1,86 @@
 require 'csv'
 require 'time'
-
 require_relative 'passenger'
 require_relative 'trip'
 
 module RideShare
   class TripDispatcher
     attr_reader :drivers, :passengers, :trips
-
+    
     def initialize(directory: './support')
       @passengers = Passenger.load_all(directory: directory)
       @trips = Trip.load_all(directory: directory)
+      @drivers = Driver.load_all(directory: directory)
       connect_trips
     end
-
+    
     def find_passenger(id)
       Passenger.validate_id(id)
       return @passengers.find { |passenger| passenger.id == id }
     end
+    
+    def find_driver(id)
+      Driver.validate_id(id)
+      return @drivers.find{ |driver| driver.id == id }
+    end
+    
+    def request_trip(passenger_id)
+      # find the passenger using their id 
+      found_passenger = @passengers.find do |passenger|
+        passenger.id == passenger_id
+      end
+      p_id = found_passenger.id # should be the same as the method's argument 
```

Comment:
You're right! It should be the same as the argument. In that case, do you need this variable assignment?